lib/openssl/asn1.rb

```diff
@@ -150,30 +275,201 @@ def each(&blk)
 
         self
       end
+
+      def to_der
+        cons_to_der
+      end
     end
 
-    class Boolean < Primitive ; end
-    class Integer < Primitive ; end
-    class Enumerated < Primitive ; end
+    class Null < Primitive
+      def to_der
+        raise ASN1Error,  "nil expected" unless @value.nil?
+
+        "\x05\x00"
+      end
```

Comment:
Null can define a `der_value` that returns an empty string, similar to other primitive encoding classes.

Reply:
I guess it could, but it seems pointless in this case, as we already know the outcome?

Reply:
It may specify tagging, e.g. `OpenSSL::ASN1::Null.new(nil, 123, :EXPLICIT).to_der`.

Reply:
fair enough, done.